Webhook Call Fails with Error "Unexpected type for 'content', <class 'dict'>"

### Description

When adding a new document, the workflow webhook intended to call the configured URL (http://<URL>/endpoint/paperlessnotification) fails. Instead of processing the webhook call correctly, the system returns the error:

`Failed attempt sending webhook to http://<URL>/endpoint/paperlessnotification: Unexpected type for 'content', <class 'dict'>`

<img width="547" alt="Image" src="https://github.com/user-attachments/assets/799e1469-c77b-4b50-ab79-f08a1415abdf" />

### Steps to reproduce

- Configure paperless-ngx with a webhook endpoint (e.g., http://<URL>/endpoint/paperlessnotification).
- Add a new document to the system.
- Observe that the webhook is triggered on document addition.
- Notice in the logs that instead of successfully sending the webhook, the error related to the unexpected type for content occurs.


### Webserver logs

```bash
[2025-04-16 08:21:12,002] [INFO] [paperless.tasks] ConsumeTaskPlugin completed with: Success. New document id 87 created
[2025-04-16 08:21:15,465] [ERROR] [paperless.handlers] Failed attempt sending webhook to http://<URL>/endpoint/paperlessnotification: Unexpected type for 'content', <class 'dict'>
```

### Browser logs

```bash

```

### Paperless-ngx version

2.15.2

### Host OS

Linux-6.10.14-linuxkit-x86_64-with-glibc2.36

### Installation method

Docker - official image

### System status

```json
{
    "pngx_version": "2.15.2",
    "server_os": "Linux-6.10.14-linuxkit-x86_64-with-glibc2.36",
    "install_type": "docker",
    "storage": {
        "total": 250685575168,
        "available": 157277913088
    },
    "database": {
        "type": "sqlite",
        "url": "/usr/src/paperless/data/db.sqlite3",
        "status": "OK",
        "error": null,
        "migration_status": {
            "latest_migration": "paperless_mail.0029_mailrule_pdf_layout",
            "unapplied_migrations": []
        }
    },
    "tasks": {
        "redis_url": "redis://broker:6379",
        "redis_status": "OK",
        "redis_error": null,
        "celery_status": "OK",
        "celery_url": "celery@13f5ed52847a",
        "celery_error": null,
        "index_status": "OK",
        "index_last_modified": "2025-04-16T08:21:11.851754+02:00",
        "index_error": null,
        "classifier_status": "OK",
        "classifier_last_trained": "2025-04-16T06:05:00.437310Z",
        "classifier_error": null,
        "sanity_check_status": "WARNING",
        "sanity_check_last_run": null,
        "sanity_check_error": "No sanity check tasks found"
    }
}
```

### Browser

_No response_

### Configuration changes

_No response_

### Please confirm the following

- [x] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [x] This issue is not about the OCR or archive creation of a specific file(s). Otherwise, please see above regarding OCR tools.
- [x] I have already searched for relevant existing issues and discussions before opening this report.
- [x] I have updated the title field above with a concise description.

--- a/src/documents/signals/handlers.py
+++ b/src/documents/signals/handlers.py
@@ -622,20 +622,30 @@ def send_webhook(
     as_json: bool = False,
 ):
     try:
+        post_args = {
+            "url": url,
+            "headers": headers,
+            "files": files,
+        }
         if as_json:
-            httpx.post(
-                url,
-                json=data,
-                files=files,
-                headers=headers,
-            ).raise_for_status()
+            post_args["json"] = data
+        elif isinstance(data, dict):
+            post_args["data"] = data
         else:
-            httpx.post(
-                url,
-                content=data,
-                files=files,
-                headers=headers,
-            ).raise_for_status()
+            post_args["content"] = data
+
+        httpx.post(
+            **post_args,
+        ).raise_for_status()
+        logger.info(
+            f"Webhook sent to {url}",
+        )
+    except Exception as e:
+        logger.error(
+            f"Failed attempt sending webhook to {url}: {e}",
+        )
+        raise e
+
         logger.info(
             f"Webhook sent to {url}",
         )

--- a/src/documents/tests/test_workflows.py
+++ b/src/documents/tests/test_workflows.py
@@ -2614,7 +2614,7 @@ class TestWorkflows(
             )
 
             mock_post.assert_called_once_with(
-                "http://paperless-ngx.com",
+                url="http://paperless-ngx.com",
                 content="Test message",
                 headers={},
                 files=None,
@@ -2622,6 +2622,20 @@ class TestWorkflows(
 
             expected_str = "Webhook sent to http://paperless-ngx.com"
             self.assertIn(expected_str, cm.output[0])
+
+            # with dict
+            send_webhook(
+                url="http://paperless-ngx.com",
+                data={"message": "Test message"},
+                headers={},
+                files=None,
+            )
+            mock_post.assert_called_with(
+                url="http://paperless-ngx.com",
+                data={"message": "Test message"},
+                headers={},
+                files=None,
+            )
 
     @mock.patch("httpx.post")
     def test_workflow_webhook_send_webhook_retry(self, mock_http):

